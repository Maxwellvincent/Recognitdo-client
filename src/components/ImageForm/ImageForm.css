.form {
    width: 700px;
    background-color:black;
    background-image:
    radial-gradient(white, rgba(255,255,255,.2) 2px, transparent 40px),
    radial-gradient(white, rgba(255,255,255,.15) 1px, transparent 30px),
    radial-gradient(white, rgba(255,255,255,.1) 2px, transparent 40px),
    radial-gradient(rgba(255,255,255,.4), rgba(255,255,255,.1) 2px, transparent 30px);
    background-size: 550px 550px, 350px 350px, 250px 250px, 150px 150px;
    background-position: 0 0, 40px 60px, 130px 270px, 70px 100px;
}

.f3 {
    font-size: 1.5rem;
}

.f4 {
    font-size:1.25rem
}

.w-70 {
    width: 70%;
}

.dib {
    display: inline-block;
}

.white { 
    color: #fff; 
}

.bg-light-purple { 
    background-color: #a463f2; 
}

.grow {
    -moz-osx-font-smoothing: grayscale; 
    -webkit-backface-visibility: hidden; 
    backface-visibility: hidden; 
    -webkit-transform: translateZ( 0 ); 
    transform: translateZ( 0 ); 
    transition: -webkit-transform .25s ease-out; 
    transition: transform .25s ease-out; 
    transition: transform .25s ease-out, -webkit-transform .25s ease-out;
}

.pa4 { 
    padding: 2rem; 
}

.br3 {
    border-radius: .5rem;
}

.shadow-5 {
    box-shadow:4px 4px 8px 0px rgba( 0, 0, 0, 0.2 );
}

.center {
    margin-right: auto;
    margin-left: auto;
}


.w-30 {
    width: 30%;
}

.link {
    text-decoration: none;
    transition: color .15s ease-in;
}

.pa2 { 
    padding: .5rem; 
}

.ph3 { 
    padding-left: 1rem; 
    padding-right: 1rem; 
}

.pv2 { 
    padding-top: .5rem; 
    padding-bottom: .5rem; 
}