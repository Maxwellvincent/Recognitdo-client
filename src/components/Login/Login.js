import React, { useState } from "react";
import { useHistory, Link } from "react-router-dom";
import "./Login.css";
import { toast } from "react-toastify";
import { useForm } from "react-hook-form";

const Login = ({setAuth }) => {
  const { register, handleSubmit, errors } = useForm();
  const history = useHistory();
  const [inputs, setInputs] = useState({
    email: "",
    password: ""
  })

  const {email, password} = inputs;

  const onChange = (e) => {
    setInputs({...inputs, [e.target.name]: e.target.value})
  }

  const onSubmit = (data,e) => {
    onSubmitLogin(e);
  };

  const onSubmitLogin = async (e) => {
    e.preventDefault();
    // http://localhost:3001/auth/login
    // https://rocky-oasis-94549.herokuapp.com/auth/login

    try {
      const body = {email, password};

      const response = await fetch("https://rocky-oasis-94549.herokuapp.com/auth/login", {
      method: "POST",
      headers: {
        "Content-Type": "application/json",
        Accept: "application/json",
      },
      body: JSON.stringify(body)
      })

    const parseRes = await response.json();

    localStorage.setItem("token", parseRes.token);

    setAuth(true);

    } catch (err) {
      console.error(err.message);
    }

  };

  return (
    <article className="br3 ba  b--black-10 mv4 w-100 w-50-m w-25-l mw6 shadow-5 center">
      <main className="pa4 black-80">
        <form
          className="measure"
          style={{ textAlign: "center" }}
          onSubmit={handleSubmit(onSubmit)}
        >
          <fieldset id="sign_up" className="ba b--transparent ph0 mh0">
            <legend className="f1 fw6 ph0 mh0">Login</legend>
            <div className="mt3">
              <label className="db fw6 lh-copy f6" htmlFor="email">
                Email
              </label>
              <input
                className="pa2 input-reset ba bg-transparent hover-bg-black hover-white w-100"
                type="email"
                name="email"
                id="email"
                required
                value={email}
                ref={register({ required: true })}
                onChange={e => onChange(e)}
              />
              {errors.email && <span>This field is required</span>}
            </div>
            <div className="mv3">
              <label className="db fw6 lh-copy f6" htmlFor="password">
                Password
              </label>
              <input
                className="b pa2 input-reset ba bg-transparent hover-bg-black hover-white w-100"
                type="password"
                name="password"
                id="password"
                required
                value={password}
                ref={register({ required: true })}
                onChange={e => onChange(e)}
              />
              {errors.password && <span>This field is required</span>}
            </div>
          </fieldset>
          <div className="">
            <input
              className="b ph3 pv2 input-reset ba b--black bg-transparent grow pointer f6 dib"
              type="submit"
              value="Login"
              onSubmit={(e) => onSubmitLogin(e)}
            />
          </div>
          <div className="lh-copy mt3">
            <Link
              to="/register"
              style={{ textDecoration: "none", cursor: "pointer" }}
            >
              <p className="f6 link dim black db pointer">Register</p>
            </Link>
          </div>
          <div className="lh-copy mt3">
            <Link to="/" style={{ textDecoration: "none", cursor: "pointer" }}>
              <p className="f6 link dim black db">Home Page</p>
            </Link>
          </div>
        </form>
      </main>
    </article>
  );
};

export default Login;
