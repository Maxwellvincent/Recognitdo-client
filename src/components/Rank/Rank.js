import React from 'react'
import './Rank.css';

const Rank = ({user}) => {
  console.log(user);
  // const {name, entries} = user.user
    return (
        <div>
            <div className='white f3'>
              {`${user.name.toUpperCase()}, you have submited `} 
            </div>
            <div className='white f1'>
              {`${user.entries} images`}  
            </div>
        </div>
    )
}

export default Rank
