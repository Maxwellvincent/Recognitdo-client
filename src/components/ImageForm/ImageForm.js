import React from "react";
import "./ImageForm.css";

function ImageForm({ onInputChange, onSubmit,getName }) {
  return (
    <div>
      <p className="f3">
        {
          "Submit an image URL that contains an individuals head in order for our Facial Recognition software to locate a face within the image. As you submit more images the counter will increase."
        }
      </p>
      <div className="center">
        <div className="form center pa4 br3 shadow-5">
          <label
            htmlFor="user-img"
            style={{
              color: "white",
              padding: ".3rem",
              display: "flex",
              width: "100%",
            }}
          >
            {" "}
            Insert Image Url:
            <input
              className="f4 pa2 w-70 center"
              type="text"
              name="user-img"
              id="user-img"
              onChange={onInputChange}
            />
          </label>
          <button
            className="w-30 grow f4 link ph3 pv2 dib white bg-light-purple"
            onClick={onSubmit}
          >
            Detect
          </button>
        </div>
      </div>
    </div>
  );
}

export default ImageForm;
