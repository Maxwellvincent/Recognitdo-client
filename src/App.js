import Clarifai from 'clarifai';
import React, {useState, useEffect} from 'react';
import { BrowserRouter as Router, Route } from "react-router-dom";
import {toast} from 'react-toastify';
import 'react-toastify/dist/ReactToastify.css';
import Nav from './components/Navigation/Navigation';
import Logo from './components/Logo/Logo';
import ImageForm from './components/ImageForm/ImageForm';
import Rank from './components/Rank/Rank';
import Particles from 'react-particles-js';
import FaceRecognition from './components/FaceRecognition/FaceRecognition';
import Signin from './components/Signin/Signin';
import Register from './components/Register/Register'
import config from './config';
import './App.css';
import LandingPage from './components/LandingPage/LandingPage';


toast.configure();
const app = new Clarifai.App({
  
  apiKey: `${config.API_KEY}`
 });


const particleOptions = {
  particles: {
    number: {
      value: 30,
      density: {
        enable: true,
        value_area: 800
      }
    }
  }
}

function App() {
  const [input, setInput] = useState('');
  const [imageUrl, setImageUrl] = useState('');
  const [box, setBox] = useState({});
  const [route, setRoute] = useState('signin');
  const [isSignedIn, setIsSignedIn] = useState(false);
  const [user, setUser] = useState({
            id: '',
            name: '',
            email: '',
            entries: 0,
            joined: ''
  });

  useEffect(() => {
    fetch('https://rocky-oasis-94549.herokuapp.com/api/')
    .then(resp => resp.json())
  })

  const calculateFaceLocation = (data) => {
    const clarafaiFace = data.outputs[0].data.regions[0].region_info.bounding_box;
    const image = document.getElementById('inputImage');
    const width = Number(image.width);
    const height = Number(image.height);
    return {
      leftCol: clarafaiFace.left_col * width,
      topRow: clarafaiFace.top_row * height,
      rightCol: width - (clarafaiFace.right_col * width),
      bottomRow: height - (clarafaiFace.bottom_row * height)
    }
  }

  const displayFaceBox = (box) => {
    setBox(box);
  }


  const onInputChange = e => {
    setInput(e.target.value)
  };

  const onSubmit = async () => {
    setImageUrl(input);
    await app.models
      .predict(
        Clarifai.FACE_DETECT_MODEL,
        input)
      .then( async response => {
        if(response){
          await fetch('https://rocky-oasis-94549.herokuapp.com/image', {
            method: 'PUT',
            headers: {'Content-Type': 'application/json'},
            body: JSON.stringify({
              id: user.id
            })
          })
          .then(async response => response.json())
          .then(async count => {
            user.entries = count;
          })
        }
          return displayFaceBox(calculateFaceLocation(response));
        })
      .catch(err => console.error(err))
  }

  const onRouteChange = (route) => {
    if(route === 'signout') {
      setIsSignedIn(false)
      
    } else if(route === 'home'){
      setIsSignedIn(true)
      
    } else if (route === 'signin'){
      setIsSignedIn(false);
    }
    setRoute(route);
  }

  const loadUser = (data) => {
    setUser({
      id: data.id,
      name: data.name,
      email: data.email,
      entries: data.entries,
      joined: data.joined,
    });
  }

  return (
    <div className="App">
      <Router>
        {/* <Particles 
          className="particles" 
          params={particleOptions}
        /> */}
        {/* <Nav isSignedIn={isSignedIn} onRouteChange={onRouteChange}/> */}
        <Route exact path="/" component={LandingPage}/>
        <Route exact path="/home" render={ props => 
            <div>
              <Nav isSignedIn={isSignedIn} onRouteChange={onRouteChange}/> 
              <Particles className="particles" params={particleOptions}/>
              <Logo/>
              <Rank user={user}/>
              <ImageForm onInputChange={onInputChange} onSubmit={onSubmit}/>
              <FaceRecognition box={box} imageUrl={imageUrl}/>
            </div>
          
          } 
        />
        <Route exact path="/signin" render={props => 
                <div>
                    <Particles className="particles" params={particleOptions}/>
                    <Signin onRouteChange={onRouteChange} loadUser={loadUser}/>
                </div>
                }
        />
        <Route exact path="/register" render={props => 
                <div>
                    <Particles className="particles" params={particleOptions}/>
                    <Register onRouteChange={onRouteChange} loadUser={loadUser} isSignedIn={isSignedIn}/>
                </div>
                }
        />
          {/* {route === 'home' 
            ? <>
                <Logo/> 
                <Rank user={user}/>
                <ImageForm onInputChange={onInputChange} onSubmit={onSubmit}/>
                <FaceRecognition box={box} imageUrl={imageUrl}/>
              </>
            : (route === 'signin' 
                ? 
                  <Signin onRouteChange={onRouteChange} loadUser={loadUser}/> 
                  : <Register onRouteChange={onRouteChange} loadUser={loadUser}/>
            )
          } */}
      </Router>
    </div>
  );

}

export default App;
