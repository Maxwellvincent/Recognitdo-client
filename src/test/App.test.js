import React from "react";
import ReactDOM from "react-dom";
import App from "../App";

import Enzyme, { shallow, render, mount } from "enzyme";
import toJson from "enzyme-to-json";
import Adapter from "enzyme-adapter-react-16";
import ImageForm from "../components/ImageForm/ImageForm";
import Login from "../components/Login/Login";
import Rank from "../components/Rank/Rank";

describe("<App/>", () => {
  it("renders the enzyme correctly", () => {
    const wrapper = shallow(<App />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it("Should render the <App /> component", () => {
    const wrapper = render(<App />);
    expect(wrapper);
  });

  it("renders the <ImageForm/>", () => {
    const wrapper = render(<ImageForm />);
    expect(wrapper);
  });

  it("renders the Rank component", () => {
    const user = { name: "drew", entries: 1 };
    const wrapper = render(<Rank user={user} />);
    expect(wrapper);
  });
});

Enzyme.configure({ adapter: new Adapter() });
