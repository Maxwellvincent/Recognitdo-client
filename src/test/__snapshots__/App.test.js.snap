// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App/> renders the App component correctly 1`] = `
<div
  className="App"
>
  <BrowserRouter>
    <Route
      component={[Function]}
      exact={true}
      path="/"
    />
    <Route
      exact={true}
      path="/dashboard"
      render={[Function]}
    />
    <Route
      exact={true}
      path="/login"
      render={[Function]}
    />
    <Route
      exact={true}
      path="/register"
      render={[Function]}
    />
  </BrowserRouter>
</div>
`;

exports[`<App/> renders the enzyme correctly 1`] = `
<div
  className="App"
>
  <BrowserRouter>
    <Route
      component={[Function]}
      exact={true}
      path="/"
    />
    <Route
      exact={true}
      path="/dashboard"
      render={[Function]}
    />
    <Route
      exact={true}
      path="/login"
      render={[Function]}
    />
    <Route
      exact={true}
      path="/register"
      render={[Function]}
    />
  </BrowserRouter>
</div>
`;
